serial_protocol.md

```diff
@@ -0,0 +1,61 @@
+#Protocol:
+During the first use, or when the chiplet changes environments a “NETWORK” call is expected to initialize the wifi parameters.
+
+Every time a serial connection is established we will expect a “INIT” call after which any subsequent calls can be made, until the connection is closed. 
+
+In the following examples we use $ to represent variables. For example $Host represents your firebase host.
+##NETWORK
+Only needs to be called when the chiplet is in a new environment and needs to connect to a new network. This setting will be stored by the chiplet and persisted through power cycles.
+###Usage
+	NETWORK $SSID
+	NETWORK $SSID $PASSWORD
+###Response
+	OK - When connected to new network
+	FAIL - When unable to connect
+###Examples
+	>> NETWORK home-guest
+	<< OK
+	>> NETWORK home-private MySecretPassword
+	<< OK
+
+##INIT
+Must be called after creating a Serial connection, it can take either just a host for accessing public variables or you may also provide a secret for accessing protected variables in the database.
+###Usage
+	INIT $Host
```

Comment:
Version changes are pretty dramatic and I guess I usually only consider them for unforeseen needs. If we have a pretty good idea we want to support other protocols and the cost is pretty minor why not just support it now?
I see it as:
```
BEGIN_FIREBASE https://...
GET /foo
PUT /bar
BEGIN_GCM
SEND message
....
```
So it is a modal connection based on the last thing you "BEGIN_"'d. Seems the only cost to support this now is having _FIREBASE at the end of the command.